### PR TITLE
H265sample

### PR DESCRIPTION
**Pull request by Tianhaol**
This files is  an example tool to encode HEVC.

### CODE REVIEW

**Comment by sreerenjb** [comment]
Hm, This is a task assigned to GSoC student! 
I have created this bug report https://github.com/intel/libva-utils/issues/107
specifically to avoid any possible collision like this. !!!!!!!!

**Comment by XinfengZhang** [comment]
@sreerenjb  are there timelines, and if there are opinion to enable low delay B frame?

**Comment by XinfengZhang** [comment]
@sreerenjb  I dont think there cant be two hevc sample, it can both exist

**Comment by sreerenjb** [comment]
@XinfengZhang Don't worry, We have assigned a different task to the student.
I request developers to be a bit more careful in future before sending PR like this.
In Open Source projects, we never know who is working on what, only way to avoid collision is to create an issue/bug_report if someone working on a feature so that others can see what is happening.

Again, it is okay this time.

**Comment by XinfengZhang** [comment]
@sreerenjb thanks a lot, yes, we should follow rule of open source world. will request more discussion  before making decision. file an issue before a PR. thanks.

**Comment by XinfengZhang** [comment]
@Tianhaol please help to rebase and clean your commits, another is about the commit description. please add some description in the commit.

File: encode/Makefile.am [hunk 1]
```diff
@@ -36,6 +36,14 @@ h264encode_LDADD	= \
 	$(top_builddir)/common/libva-display.la \
 	-lpthread -lm
 
+hevcencode_SOURCES	= hevcencode.c
+hevcencode_CFLAGS	= -I$(top_srcdir)/common -g
+hevcencode_LDADD	= \
+       	$(LIBVA_LIBS) \
+	$(top_builddir)/common/libva-display.la \
```
Comment (cchunbo) [inline encode/Makefile.am]: better to keep alignment with previous

File: encode/hevcencode.c [hunk 2]
```diff
@@ -0,0 +1,3425 @@
...
+};
+
+// Config const values
+#define MAX_TEMPORAL_SUBLAYERS         8
+#define MAX_LAYER_ID                   64
+#define MAX_LONGTERM_REF_PIC           32
+#define NUM_OF_EXTRA_SLICEHEADER_BITS  3
+struct ProfileTierParamSet
+{
+    uint8_t      general_profile_space;                                        //u(2)
```
Comment (cchunbo) [inline encode/hevcencode.c]: is it better to use bit field for the params ?

File: encode/Makefile.am (hunk 1 above)
Comment (Tianhaol) [reply encode/Makefile.am]: Hevc has the same alignment with h264.
Could you tell me which line is different?

**Pull request merged**